### ISSUE TITLE
Adding the same key twice causes an error, though documentation claims it should not

### ISSUE THREAD

**Issue by drPytho** [lestrrat-go/jwx#730]
**Contribution Guidelines**

Before filing an issue, please read the contents of [CONTRIBUTING.md](https://github.com/lestrrat-go/jwx/blob/v2/.github/CONTRIBUTING.md), and follow its instructions.

**Describe the bug**

The documentation claims that [adding the same key twice will not generate an error](https://github.com/lestrrat-go/jwx/blob/v2.0.0/jwk/interface.go#L52=), however looking at the code for [AddKey](https://github.com/lestrrat-go/jwx/blob/v2.0.0/jwk/set.go#L88=) it does return an error if the key already exists. 

Go 1.18

**To Reproduce / Expected behavior**
https://go.dev/play/p/zCddGgFzF1f

```go
package main

import (
	"fmt"

	"github.com/lestrrat-go/jwx/v2/jwk"
)

func main() {
	set := jwk.NewSet()
	key, err := jwk.FromRaw([]byte("someRandomString"))
	if err != nil {
		fmt.Println("Failed to create key", err)
	}

	if err := set.AddKey(key); err != nil {
		fmt.Println("First attempt", err)
	}

	if err := set.AddKey(key); err != nil {
		fmt.Println("Second attempt", err)
	}
}
```

**Comment by lestrrat** [comment]
Thanks! That's a documentation bug for v2. Will add in Changes-v2.md, and fix the documentation in for `type Set`

**Comment by lestrrat** [comment]
fixed via #731
